internal/services/vorpal.go

```diff
@@ -93,16 +93,22 @@ func executeScan(vorpalWrapper grpcs.VorpalWrapper, filePath string) (*grpcs.Sca
 	return vorpalWrapper.Scan(fileName, sourceCode)
 }
 
-func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper) error {
+func manageVorpalInstallation(vorpalParams VorpalScanParams, vorpalWrapper grpcs.VorpalWrapper, vorpalWrappers VorpalWrappersParam) error {
 	vorpalInstalled, _ := osinstaller.FileExists(vorpalconfig.Params.ExecutableFilePath())
 
-	if vorpalParams.VorpalUpdateVersion || !vorpalInstalled {
-		if err := vorpalWrapper.HealthCheck(); err == nil {
-			_ = vorpalWrapper.ShutDown()
+	if !vorpalInstalled || vorpalParams.VorpalUpdateVersion {
+		if err := checkLicense(vorpalParams.IsDefaultAgent, vorpalWrappers); err != nil {
```

Comment:
I think Or told us to shut down the service here

Reply:
we need to check here for helth check before shutting down. or the shut down check that there sis engine alive bofre>?